### DISCUSSION TITLE
Frigate 100% CPU and Memory without apparent reason

### DISCUSSION THREAD

**Discussion by Claudio1L** [tteck/Proxmox#3276] [Help Needed]
HI

Frigate goes in stuck sometimes without apparent reasons.
It's only my issue?
Have to hard stop the LXC container and restart

![immagine](https://github.com/tteck/Proxmox/assets/97904190/83560b1b-fd6c-4e73-8a08-4f01f5dc7d31)
![immagine](https://github.com/tteck/Proxmox/assets/97904190/31e20f9e-e964-460c-941e-3ab5eb31590a)

**Comment by Mellowlynx** [comment]
I only have seen this happen with Home Assistant from time to time. 
Rebooting after an OS update and backup stopped seems to stop this behavior, or at least it happens less often.

When it happens again, try to run `top` or something similar to see what the system is doing, it might give a clue.

**Reply by Claudio1L** [reply]
yes have seen that on homeassistant core on LXC, but only happen to me if the internet connection or the routing towards the wan goes down. I'm investigating if the cause is any custom integration, because have another instance that does not suffer of the same issue. If I spin up a new container and copy the config, the issue follow the container.

**Reply by Claudio1L** [reply]
at the moment I have another HA instance with the proxmox integration, monitoring the LXC on the main instance that when Frigate or HA goes 100% cpu restart the containers

**Reply by Mellowlynx** [reply]
I monitor it too (through Grafana) and will check the WAN connection when I have it happen again.

- I first thought It was an add-on, but docker was running normal the last time I saw it.
Thought I had a screenshot, but can't fine it. I will share it when it happens again.

**Reply by Claudio1L** [reply]
Are you using HA core directly on LXC, or HA core on docker on LXC ?  Just to know if the issue happen also on docker

**Reply by Mellowlynx** [reply]
I use the Home Assistant OS VM

**Reply by Claudio1L** [reply]
Ah ok, so different setup from mine. Similar issue. Let me lnow if you find something

**Comment by Chrischi-** [comment]
Do you use opus audio for streams (go2rtc config)? If so, then deactivate audio completely.

If not, a test with pure video streams (no audio) would still make sense.

Bug should be fixed with frigate 14.x

**Comment by willgryan** [comment]
I experience this occasionally as well. Proxmox LXC on Intel N100 w/ Coral TPU. Frigate 0.13.2, go2rtc 1.9.4. I'm not using opus audio (h264 video/pcm_alawa audio). 

I can't see if any process is hanging because the container freezes, requiring a force stop on the host. I thought maybe it was an OOM issue, because the script's default is 1GB, but then I noted an 8GB ram total in the original commenter's container. 

It's interesting to note that, when my container was hanging, I increased the ram from 1GB>2GB in the Proxmox GUI and it instantly resolved the issue, and Frigate was resolvable again.

**Reply by willgryan** [reply]
I'm still experiencing this issue. When I increase the container memory, the ffmpeg process consumes it all. The ffmpeg process also no longer shows on the host with intel_gpu_top.

Interestingly, I run [restreamer ](https://github.com/datarhei/restreamer/) in a docker LXC which uses ffmpeg streams without hardware acceleration, and it occasionally experiences this same issue. 

This leads me to believe it's not a hardware acceleration problem, but an ffmpeg problem. 

```
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root           1  0.0  0.0 165400  2816 ?        Ss   Aug30   1:31 /sbin/init
root          56  0.0  0.0  31816   916 ?        Ss   Aug30   0:41 /lib/systemd/systemd-journald
systemd+      69  0.0  0.0  15868  1536 ?        Ss   Aug30   1:29 /lib/systemd/systemd-networkd
root          91  0.0  0.0  99768   592 ?        Ssl  Aug30   0:00 /sbin/dhclient -4 -v -i -pf /run/dhclient.eth0.pid -lf /var/lib/dhcp/dhclient.eth0.leases -I -df /var/lib/dhcp/dhclient6.eth0.leases eth0
_rpc          99  0.0  0.0   7600   128 ?        Ss   Aug30   0:00 /sbin/rpcbind -f -w
systemd+     107  0.0  0.0  23520   868 ?        Ss   Aug30   0:19 /lib/systemd/systemd-resolved
message+     123  0.0  0.0   7992   256 ?        Ss   Aug30   0:00 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
root         127  0.0  0.0 220548   384 ?        Ssl  Aug30   0:00 /usr/sbin/rsyslogd -n -iNONE
root         130  0.0  0.0  13220   384 ?        Ss   Aug30   0:25 /lib/systemd/systemd-logind
root         146  0.0  0.1 1240744 12192 ?       Ssl  Aug30  74:19 /usr/local/go2rtc/bin/go2rtc -config=/dev/shm/go2rtc.yaml
root         148  0.0  0.5 2725200 47180 ?       Ssl  Aug30  49:39 python3 -u -m frigate
root         153  0.0  0.0 693972   460 ?        Ss   Aug30   0:00 nginx: master process /usr/local/nginx/sbin/nginx
root         162  0.0  0.0 957248   676 ?        Sl   Aug30   0:00 nginx: worker process
root         163  0.0  0.0 957248   420 ?        Sl   Aug30   0:00 nginx: worker process
root         164  0.0  0.0 956984   292 ?        Sl   Aug30   0:00 nginx: worker process
root         230  0.0  0.0 956984   548 ?        Sl   Aug30   0:00 nginx: worker process
root         244  0.0  0.0 694400   804 ?        S    Aug30   1:22 nginx: cache manager process
root         327  0.0  0.0   3624  1152 ?        Ss   Aug30   0:49 /usr/sbin/cron -f
root         342  0.0  0.0   2512   384 pts/0    Ss+  Aug30   0:00 /sbin/agetty -o -p -- \u --noclear --keep-baud console 115200,38400,9600 linux
root         345  0.0  0.0   7008   384 pts/1    Ss   Aug30   0:00 /bin/login -f
root         349  0.0  0.0   2512   256 pts/2    Ss+  Aug30   0:00 /sbin/agetty -o -p -- \u --noclear --keep-baud tty2 115200,38400,9600 linux
root         466  0.0  0.0  39804   776 ?        Ss   Aug30   0:18 /usr/lib/postfix/sbin/master -w
postfix      468  0.0  0.0  40104   128 ?        S    Aug30   0:00 qmgr -l -t unix -u
root         473  0.0  0.0   5160  1920 pts/1    S    Aug30   0:31 -bash
root         479  0.0  0.0 684560  6548 ?        S    Aug30   2:51 frigate.logger
root         481  0.0  0.1 883792 10044 ?        Sl   Aug30   8:21 frigate.recording_manager
root         489  0.0  0.0  17496  1920 ?        S    Aug30   6:05 /usr/bin/python3 -c from multiprocessing.resource_tracker import main;main(46)
root         492  0.0  0.1 1396700 8772 ?        Sl   Aug30  28:58 frigate.output
root         497  0.0  0.2 1181704 17596 ?       Sl   Aug30  25:23 frigate.process:side
root         498  0.0  0.0 126820   512 ?        Ss   Aug30   0:00 ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 640x360 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:
root         499  0.0  0.9 1221052 76916 ?       Sl   Aug30 179:10 frigate.process:front
root         510  0.0  0.3 1186020 33112 ?       Sl   Aug30 136:39 frigate.process:back
root         511  0.0  0.0 126820   384 ?        Ss   Aug30   0:00 ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 640x360 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:
root         514  0.0  0.1 1172996 13500 ?       Sl   Aug30  10:56 frigate.capture:side
root         516  0.0  0.0 126820   256 ?        Ss   Aug30   0:00 ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 640x360 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:
root         517  0.0  0.0 1173080 6176 ?        Sl   Aug30  21:12 frigate.capture:front
root         522  0.0  0.1 1172740 12504 ?       Sl   Aug30  24:06 frigate.capture:back
root         529  0.0  0.0 126820   256 ?        Ss   Aug30   0:00 ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:
**_root       80608  0.0 86.3 7878848 7157476 ?     Ssl  04:18   6:00 ffmpeg -hide_banner -loglevel warning -threads 2 -hwaccel_flags allow_profile_mismatch -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -hwaccel_output_format vaa_**
postfix    87611  0.0  0.0  40056  1024 ?        S    11:57   0:28 pickup -l -t unix -u -c
root       88813  0.0  0.7 2732084 61436 ?       Sl   13:18   1:27 frigate.detector.coral1
root       88853  0.0  0.0   6496  2688 pts/1    R+   13:28   0:00 ps aux
```
